tmk_core/common/action.c

```diff
@@ -55,7 +55,7 @@ void action_exec(keyevent_t event)
 
 #if !defined(NO_ACTION_LAYER) && defined(PREVENT_STUCK_MODIFIERS)
 bool disable_action_cache = false;
-int8_t pressed_actions_cache[MATRIX_ROWS][MATRIX_COLS];
+uint8_t source_layers_cache[5][((MATRIX_ROWS * MATRIX_COLS) / 8) ? ((MATRIX_ROWS * MATRIX_COLS) / 8) : 1];
```

Comment:
I still think it won't work for keyboards with key count not divisible by 8. `(MATRIX_ROWS * MATRIX_COLS) / 8` will round down/truncate the result and we won't have enough array elements for all the keys.

Consider the following modification:

    uint8_t source_layers_cache[5][((MATRIX_ROWS * MATRIX_COLS) % 8 == 0) ? ((MATRIX_ROWS * MATRIX_COLS) / 8) : ((MATRIX_ROWS * MATRIX_COLS) / 8) + 1];

Or even this one:

    uint8_t source_layers_cache[5][(MATRIX_ROWS * MATRIX_COLS + 7) / 8];


Does it make sense?